pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -219,6 +219,12 @@ func (r *REST) Update(ctx apirequest.Context, tagName string, objInfo rest.Updat
 		return nil, false, kapierrors.NewConflict(imageapi.Resource("imagestreamtags"), istag.Name, fmt.Errorf("another caller has updated the resource version to %s", imageStream.ResourceVersion))
 	}
 
+	// When we began returning image stream labels in 3.6, old clients that didn't need to send labels would be
+	// broken on update. Explicitly default labels if they are unset.  We don't support mutation of labels on update.
```

Comment:
Yeah, old clients were getting immutable errors (because they previously did not need to send labels even if the image stream had them).